lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescaleset.py

```diff
@@ -637,14 +653,22 @@ def exec_module(self, **kwargs):
                     vmss_dict['zones'] = self.zones
 
                 nicConfigs = vmss_dict['properties']['virtualMachineProfile']['networkProfile']['networkInterfaceConfigurations']
+
                 backend_address_pool = nicConfigs[0]['properties']['ipConfigurations'][0]['properties'].get('loadBalancerBackendAddressPools', [])
+                backend_address_pool += nicConfigs[0]['properties']['ipConfigurations'][0]['properties'].get('applicationGatewayBackendAddressPools', [])
+                lb_or_ag_id = None
                 if (len(nicConfigs) != 1 or len(backend_address_pool) != 1):
                     support_lb_change = False  # Currently not support for the vmss contains more than one loadbalancer
                     self.module.warn('Updating more than one load balancer on VMSS is currently not supported')
                 else:
-                    load_balancer_id = "{0}/".format(load_balancer.id) if load_balancer else None
+                    load_balancer
+                    if load_balancer:
+                        lb_or_ag_id = "{0}/".format(load_balancer.id)
+                    elif app_gateway:
+                        lb_or_ag_id = "{0}/".format(app_gateway.id)
+
                     backend_address_pool_id = backend_address_pool[0].get('id')
-                    if bool(load_balancer_id) != bool(backend_address_pool_id) or not backend_address_pool_id.startswith(load_balancer_id):
+                    if bool(lb_or_ag_id) != bool(backend_address_pool_id) or not backend_address_pool_id.startswith(lb_or_ag_id):
                         differences.append('load_balancer')
```

Comment:
difference append 'lb or  ag'?

Reply:
I just attached it to existing logic, so there can be only one (ag is also load balancer).
otherwise things will go completely out of control ;-)
or we should allow both?
in fact it's possible to have both in the same ipConfiguration, they will share subnet.
but I think the logic to make them exclusive now is ok.